python/cudf_polars/cudf_polars/containers/dataframe.py

```diff
@@ -304,36 +349,47 @@ def with_columns(
         If column names overlap, newer names replace older ones, and
         appear in the same order as the original frame.
         """
+        stream = stream or self.stream
         new = {c.name: c for c in columns}
         if replace_only and not self.column_names_set.issuperset(new.keys()):
             raise ValueError("Cannot replace with non-existing names")
-        return type(self)((self.column_map | new).values())
+        return type(self)((self.column_map | new).values(), stream=stream)
 
     def discard_columns(self, names: Set[str]) -> Self:
         """Drop columns by name."""
-        return type(self)(column for column in self.columns if column.name not in names)
+        return type(self)(
+            (column for column in self.columns if column.name not in names),
+            stream=self.stream,
+        )
 
     def select(self, names: Sequence[str] | Mapping[str, Any]) -> Self:
         """Select columns by name returning DataFrame."""
         try:
-            return type(self)(self.column_map[name] for name in names)
+            return type(self)(
+                (self.column_map[name] for name in names), stream=self.stream
+            )
         except KeyError as e:
             raise ValueError("Can't select missing names") from e
 
     def rename_columns(self, mapping: Mapping[str, str]) -> Self:
         """Rename some columns."""
-        return type(self)(c.rename(mapping.get(c.name, c.name)) for c in self.columns)
+        return type(self)(
+            (c.rename(mapping.get(c.name, c.name)) for c in self.columns),
+            stream=self.stream,
+        )
 
     def select_columns(self, names: Set[str]) -> list[Column]:
         """Select columns by name."""
         return [c for c in self.columns if c.name in names]
 
     def filter(self, mask: Column) -> Self:
         """Return a filtered table given a mask."""
-        table = plc.stream_compaction.apply_boolean_mask(self.table, mask.obj)
+        table = plc.stream_compaction.apply_boolean_mask(
+            self.table, mask.obj, stream=self.stream
```

Comment:
Is it possible that `self.table` was created on a different stream from the `mask`? If so, we still need to be ordered wrt mask's stream, right?

Reply:
*At the moment*, all uses of `DataFrame.filter` (https://github.com/rapidsai/cudf/blob/c6088d5fb5d05a9951c1d98f0e2a50137e617422/python/cudf_polars/cudf_polars/dsl/ir.py#L807, https://github.com/rapidsai/cudf/blob/c6088d5fb5d05a9951c1d98f0e2a50137e617422/python/cudf_polars/cudf_polars/dsl/ir.py#L2493) are from a `mask` that was created in `broadcast` from the `DataFrame`. Assuming our primary rule of "all data in a `DataFrame` is valid on `stream`" is met, then we're good here.

New usage of `.filter` will also need to respect this rule. I'll leave a note in the docs.